### PR TITLE
Feat/user info

### PR DESCRIPTION
**Pull request by QKKQQK**

### CODE REVIEW

**Comment by yclee0210** [comment]
Can one of the admins verify this patch?

File: middlewares/users.middleware.js [hunk 1]
```diff
@@ -203,22 +215,31 @@ function usersMiddleware(datastore, errorResponse, secret, crypto, CONFIG) {
...
             data: {
                 bid: {},
                 wishlist: [],
-                access: {},
+                access: {
+                	customer_next_payment_date: access_date,
+                	vendor_next_payment_date: access_date,
+                	customer_payment_amount: 500,
+                	vendor_payment_amount: 500,
+                	vendor_additional_paid: false
```
Comment (yclee0210) [inline middlewares/users.middleware.js:232]: I don't think this is a boolean property. We can have multiple additional payments

File: middlewares/users.middleware.js [hunk 2]
```diff
@@ -203,22 +215,31 @@ function usersMiddleware(datastore, errorResponse, secret, crypto, CONFIG) {
 
     function createUser(req, res, next) {
         const password = hashPassword(req.body.password);
-
+        var access_date = new Date();
```
Comment (yclee0210) [inline middlewares/users.middleware.js:218]: Is this a mock?

File: middlewares/users.middleware.js [hunk 3]
```diff
@@ -56,6 +57,17 @@ function usersMiddleware(datastore, errorResponse, secret, crypto, CONFIG) {
             })
     }
 
+    function getUserInfo(req, res) {
```
Comment (yclee0210) [inline middlewares/users.middleware.js:60]: maybe change to returnUserInfo?

**Review by yclee0210 — CHANGES_REQUESTED** [review]

File: middlewares/users.middleware.js (hunk 2 above)
Comment (QKKQQK) [reply middlewares/users.middleware.js:218]: It's for setting default access.vendor/customer next payment date, which is used to check access

File: middlewares/users.middleware.js (hunk 1 above)
Comment (QKKQQK) [reply middlewares/users.middleware.js:232]: I think we are not doing multiple credit, its just a check for creating an additional listing. If you change it to number then we need to provide more functionality like checking credits, buying additional credit etc. Right now the only way they get new credit is by creating additional listings, if for some reason the payment went through but creating the listing failed, he will have an extra credit, and we have to [tell him he has an existing credit] or [skip payment automatically], in both case a boolean is enough

**Review by yclee0210 — APPROVED** [review]

**Pull request merged**